driver.go

```diff
@@ -17,19 +17,33 @@
 package mysql
 
 import (
+	"context"
 	"database/sql"
 	"database/sql/driver"
+	"errors"
 	"net"
 )
 
+var (
+	errInvalidUser   = errors.New("invalid Connection: User is not set or longer than 32 chars")
+	errInvalidAddr   = errors.New("invalid Connection: Addr config is missing")
+	errInvalidNet    = errors.New("invalid Connection: Only tcp is valid for Net")
+	errInvalidDBName = errors.New("invalid Connection: DBName config is missing")
```

Comment:
Maybe, "invalid config" is better than "invalid Connection".

Reply:
I would concur and will make that change.